econml/utilities.py

```diff
@@ -639,6 +697,41 @@ def _sampled_inputs(self, X, y, sample_weight):
         data_indices = np.random.choice(X.shape[0], size=data_length, p=normalized_weights)
         return X[data_indices], y[data_indices]
 
+class WeightedLasso(Lasso):
```

Comment:
This implementation of the weightedlasso has problems with cloning. It needs to be modified to the following:
```python
class WeightedLasso(Lasso):
    def __init__(self, alpha=1.0, fit_intercept=True, precompute=False, 
                 copy_X=True, max_iter=1000, tol=0.0001, warm_start=False, positive=False,
                 random_state=None, selection='cyclic'):
        """
        TODO. Add option to normalize. Currently normalization is disabled because normalization
        with sample weights is tricky.
        """
        super().__init__(alpha=alpha, fit_intercept=fit_intercept, normalize=False, precompute=precompute, 
                 copy_X=copy_X, max_iter=max_iter, tol=tol, warm_start=warm_start, positive=positive,
                 random_state=random_state, selection=selection)
    def fit(self, X, y, sample_weight=None):
        if sample_weight is None:
            sample_weight = np.ones(X.shape[0])
        normalized_weights = sample_weight * X.shape[0] / np.sum(sample_weight)
        sqrt_weights = np.sqrt(normalized_weights)
        if self.fit_intercept:
            mean_y = np.average(y, weights=normalized_weights, axis=0)
            mean_X = np.average(X, weights=normalized_weights, axis=0)
            if ndim(y) >= 2:
                y = y - mean_y.reshape(1, -1)
            else:
                y = y - mean_y
            X = X - mean_X.reshape(1, -1)
        
        fit_intercept_temp = self.fit_intercept
        self.fit_intercept = False
        if ndim(y) >= 2:
            super().fit(X * sqrt_weights.reshape(-1, 1), y * sqrt_weights.reshape(-1,1))
        else:
            super().fit(X * sqrt_weights.reshape(-1, 1), y * sqrt_weights)
        self.fit_intercept = fit_intercept_temp
        if self.fit_intercept:
            self.intercept_ = mean_y - self.coef_ @ mean_X 
        return self
```

Reply:
Also this weighted lasso class has not been properly unit-tested. Unit tests need to be added to integrate it in master.

Reply:
Created #103 to deal with this in a separate PR.

Reply:
@kbattocchi We need to remove all the weighted lasso related stuff from this PR/branch.

Reply:
Leaving it here since Miruna has revised it and added tests.